### PR TITLE
feat: remove refetchPage

### PR DESCRIPTION
**Pull request by arnaudbzn**
Breaking change related to PR #4984 
`refetchPage` introduced is removed in favor of the new`maxPages` option

- [x] Remove `refetchPage` related code
- [x] Remove related tests
- [x] Update docs
- [x] Update migration guide breaking changes section
- [x] Update migration guide new featureq section (maxPages)
- [x] Fix types

### CODE REVIEW

File: packages/query-core/src/types.ts [hunk 1]
```diff
@@ -338,31 +338,17 @@ export interface ResultOptions {
...
 
-export interface RefetchQueryFilters<TPageData = unknown>
-  extends QueryFilters,
-    RefetchPageFilters<TPageData> {}
+export interface RefetchQueryFilters extends QueryFilters {}
 
-export interface ResetQueryFilters<TPageData = unknown>
-  extends QueryFilters,
-    RefetchPageFilters<TPageData> {}
+export interface ResetQueryFilters extends QueryFilters {}
```
Comment (TkDodo) [inline packages/query-core/src/types.ts:351]: do we need those extra interfaces if they are now exactly the same as `QueryFilters` ? I think I only made them for that feature ...

**Review by TkDodo — APPROVED** [review]

File: packages/query-core/src/types.ts (hunk 1 above)
Comment (arnaudbzn) [reply packages/query-core/src/types.ts:351]: Sure let me remove this one.

Comment (arnaudbzn) [reply packages/query-core/src/types.ts]: Done in #b338f7d5f163edc313ab9b44c682ab2aa01f3440
Still some type checks failed in vue tests..

Comment (arnaudbzn) [reply packages/query-core/src/types.ts]: All checks have passed.

**Comment by arnaudbzn** [comment]
@TkDodo Migration guide updated with a "New Features" section.

**Review by TkDodo — APPROVED** [review]

**Comment by cah4a** [comment]
Thank you for your great work. 
What if I still want to refetch just one page?

**Comment by TkDodo** [comment]
> What if I still want to refetch just one page?

there's no good way to make that work

**Pull request merged**